### ISSUE TITLE
Bug in periodic job summaries: pending never decreases

### ISSUE THREAD

**Issue by schmichael** [hashicorp/nomad#9875]
Periodic job summaries in status pending fail to decrease.

## To Reproduce

Steps to reproduce the behavior.

> Run attached periodic job.

## Expected behavior

Decremented pending status count.

## Environment:

Product Version: 1.0.0

## Repro

```hcl
job "j5" {
    datacenters = [ "dc1" ]
    type = "batch"

    periodic {
        cron = "*/1 * * * * *"
        prohibit_overlap = false
        time_zone = "UTC"
    }

    group "g5" {
        network {
          port "p5" {
            to = 6379
          }
        }
        task "t5" {
            driver = "docker"

            config {
                image = "redis:3.2"
                ports = ["p5"]
            }

        }
    }
}
```

**Comment by drewbailey** [comment]
@schmichael mind reproducing on `main`, should be covered by https://github.com/hashicorp/nomad/pull/9768

**Comment by drewbailey** [comment]
Looks like this is actually still an issue, we will at least have a failing e2e test for with #9768

**Comment by drewbailey** [comment]
fixed in #9884
